packages/rum-core/src/common/utils.js

```diff
@@ -34,33 +33,45 @@ function isCORSSupported() {
   return 'withCredentials' in xhr
 }
 
+/**
+ * Original source is from uuid module to strip `-` in the UUID generation logic
+ * https://github.com/uuidjs/uuid/blob/8977966d0061cca33a01a88f5b4893d3304d4840/src/bytesToUuid.js
+ */
 var byteToHex = []
 for (var i = 0; i < 256; ++i) {
   byteToHex[i] = (i + 0x100).toString(16).substr(1)
 }
 
-function bytesToHex(buf, offset) {
-  var i = offset || 0
-  var bth = byteToHex
-  // join used to fix memory issue caused by concatenation: https://bugs.chromium.org/p/v8/issues/detail?id=3175#c4
```

Comment:
The memory issue mentioned here is not relevant for us as we are dealing with 16 items in the array and not thousands as explained in the issue. Also, we are still using array.join for concatenation just to keep us safe from the above issue.